src/papyrus-lang-vscode/src/features/commands/SubstitutionCommands.ts

```diff
@@ -0,0 +1,35 @@
+import { commands, Disposable } from 'vscode';
+import { IExtensionConfigProvider } from '../../ExtensionConfigProvider';
+import { take } from 'rxjs/operators';
+
+export class SubstitutionCommands implements Disposable {
+    private readonly _configProvider: IExtensionConfigProvider;
+    private _importsCommand: Disposable;
+    private _sourcesCommand: Disposable;
+
+    constructor(@IExtensionConfigProvider configProvider: IExtensionConfigProvider) {
+        this._configProvider = configProvider;
+        this.registerCommands();
+    }
+
+    private async registerCommands() {
+        const config = (await this._configProvider.config.pipe(take(1)).toPromise())['skyrimSpecialEdition'];
+
+        this._importsCommand = commands.registerCommand("papyrus.projectImports", () => {
+            return config.importDirs.join(';').replace('${installPath}', config.installPath);
+        });
+
+        this._sourcesCommand = commands.registerCommand("papyrus.projectSources", () => {
+            return config.sourceDirs.join(' ').replace('${installPath}', config.installPath);
+        });
+
+        this._sourcesCommand = commands.registerCommand("papyrus.projectSources0", () => {
```

Comment:
Assigning to the same member twice. Prefer using `readonly` modifier on private class members in order to catch this.

Reply:
Ok fixed this. But this is the problem I was running into that was keeping me from using readonly. (Needed to await on config but couldn't do it in constructor.) Suggestions?

![image](https://user-images.githubusercontent.com/778287/67537814-037ee880-f6ab-11e9-9744-7b3c7bf7b656.png)

Reply:
Nah you're right, forgot this wasn't in the constructor.